starter_code/src/movies.js

```diff
@@ -1,19 +1,138 @@
 /* eslint no-restricted-globals: 'off' */
 
-// Iteration 1: All rates average - Get the average of all rates with 2 decimals 
+// Iteration 1: All rates average - Get the average of all rates with 2 decimals
+
+function ratesAverage(arr) {
+    const avg =
+        arr.reduce(function (accumulator, movie) {
+            return accumulator + parseFloat(movie.rate);
+        }, 0) / arr.length;
+    return parseFloat(avg.toFixed(2));
+}
 
- 
 // Iteration 2: Drama movies - Get the average of Drama Movies
 
+function dramaMoviesRate(arr) {
+    const dramaFilms = arr.filter(function (movie) {
+        if (movie.genre.includes("Drama")) {
+            return true;
+        }
+    });
+
+    if (dramaFilms.length === 0) {
+        return 0;
+    }
+
+    const dramaAvg =
```

Comment:
Hey Alex!
In this case you basically took the ratesAverage function you did in the previous iteration. Ideally, you could have used that same function :) Something like ratesAverage(dramaFilms). With this you would have saved some time and keystrokes.
Remember the DRY principle in coding - Don't Repeat Yourself :) But very nice here